test/extended/openstack/loadbalancers.go

```diff
@@ -141,14 +144,115 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb] The Openstack
+			svcIp := svc.Status.LoadBalancer.Ingress[0].IP
+			svcPort := fmt.Sprint(svc.Spec.Ports[0].Port)
+
+			g.By("Checks from openshift perspective")
+			o.Expect(svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]).ShouldNot(o.BeEmpty(),
+				"load-balancer-id annotation missing")
+			loadBalancerId := svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]
+			o.Expect(svc.Status.LoadBalancer.Ingress[0].IP).ShouldNot(o.BeEmpty(), "FIP missing on svc Status")
+			o.Expect(svc.Spec.ExternalTrafficPolicy).Should(o.Equal(v1.ServiceExternalTrafficPolicyTypeLocal),
+				"Unexpected ExternalTrafficPolicy on svc specs")
+
+			g.By("Checks from openstack perspective")
+			lb, err := octavialoadbalancers.Get(loadBalancerClient, loadBalancerId).Extract()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(lb.Provider).Should(o.Equal(strings.ToLower(lbProviderUnderTest)), "Unexpected provider in the Openstack LoadBalancer")
+			pool, err := pools.Get(loadBalancerClient, lb.Pools[0].ID).Extract()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(pool.Protocol).Should(o.Equal("UDP"), "Unexpected protocol on Openstack LoadBalancer Pool: %q", pool.Name)
+			monitor, err := octaviamonitors.Get(loadBalancerClient, pool.MonitorID).Extract()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(monitor.AdminStateUp).Should(o.BeTrue(), "Unexpected healthmonitor adminStateUp on Openstack LoadBalancer Pool: %q", pool.Name)
+			o.Expect(monitor.Delay).Should(o.Equal(monitorDelay), "Unexpected healthmonitor delay on Openstack LoadBalancer Pool: %q", pool.Name)
+			o.Expect(monitor.Timeout).Should(o.Equal(monitorTimeout), "Unexpected healthmonitor timeout on Openstack LoadBalancer Pool: %q", pool.Name)
+			o.Expect(monitor.MaxRetries).Should(o.Equal(monitorMaxRetries), "Unexpected healthmonitor MaxRetries on Openstack LoadBalancer Pool: %q", pool.Name)
+			lbMethod, err := getClusterLoadBalancerSetting("lb-method", cloudProviderConfig)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(strings.ToLower(pool.LBMethod)).Should(o.Equal(lbMethod), "Unexpected LBMethod on Openstack Pool: %q", pool.LBMethod)
+
+			g.By("accessing the service 100 times from outside and storing the name of the pods answering")
+			results := make(map[string]int)
+			for i := 0; i < 100; i++ {
+				// https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/README.md#netexec
+				podName, err := getPodNameThroughUdpLb(svcIp, svcPort, "hostname")
+				if err != nil {
+					e2e.Logf("Error detected while accessing the LoadBalancer service on try %d: %q", i, err)
+				} else {
+					results[podName]++
+				}
+			}
+			e2e.Logf("Pods accessed after 100 UDP requests:\n%v\n", results)
+			pods, err := oc.KubeClient().CoreV1().Pods(oc.Namespace()).List(context.Background(), metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			//lbMethod can be something different to ROUND_ROBIN as monitors && ETP:Local are enabled on this test:
```

Comment:
do we need to check something specific around the use of ETP:local and the Pods?
It seems only the lbMethods are being checked

Reply:
mmm I don't see more checks to add here.

On line 209 we are checking that the spec is there, and we are checking also that the LbMethod different to ROUND_ROBIN is applied: In D/S CI, we are enabling monitors and setting SOURCE_IP LbMethod. Under these circunstances, a unique pod should be accesed. That is checked on line 335.

That will confirm that the ETP:Local is in place.